py/desispec/scripts/proc_tilenight.py

```diff
@@ -180,6 +182,10 @@ def main(args=None, comm=None):
             log.info(f'prestdstar failed with {error_count_pre} errors, skipping rest of tilenight steps')
         continue_tilenight=False
 
+    #- Cleanup GPU memory and rank assignments before continuing
+    desispec.gpu.free_gpu_memory()
+    desispec.gpu.redistribute_gpu_ranks(comm)
```

Comment:
There is a risk to clearing the wrong device memory pool if these functions are called in opposite order. Perhaps it would be worth adding an opt-out option like `desispec.gpu.redistribute_gpu_ranks(..., free_gpu_memory=True)` to call to `desispec.gpu.free_gpu_memory()` before switching to a new device ?